app/src/org/commcare/views/widgets/MediaWidget.java

```diff
@@ -162,4 +180,117 @@ public void setFocus(Context context) {
                 (InputMethodManager)context.getSystemService(Context.INPUT_METHOD_SERVICE);
         inputManager.hideSoftInputFromWindow(this.getWindowToken(), 0);
     }
+
+    @Override
+    public void setBinaryData(Object binaryuri) {
```

Comment:
`setBinaryData` is almost same as the earlier implementation apart from the fact that we now might have already made a copy of file at the destination in `createFilePath` (in case we were only able to get the uri from file provider and not filepath). So we are now checking if the file already exists at `destMediaPath` and only copy the flle if it doesn't (which would be the case when we would have successfully got filePath from file provider)